R/03-annotate.R

```diff
@@ -17,46 +19,196 @@
+    annotation_df <- read_table(annotation_file)
+  } else {
+    annotation_df <- get_example_data("annotation")
   }
 
-  # We'll take a look at the gimap_dataset$pg_ids and see what kinds of gene ids are there
-  # If we need to do gene conversion we'd do something like:
-
-  # biocLite('org.Hs.eg.db')
-  # mapIds(org.Hs.eg.db, <column of gene IDs>, 'ENTREZID', 'SYMBOL')
-  # https://github.com/FredHutch/GI_mapping/blob/main/workflow/scripts/02-get_pgRNA_annotations.Rmd
-
-  # This file is from https://depmap.org/portal/download/all/ and from DepMap Public 19Q3 All Files
-  # Essential gene labeling is from inst/extdata/Achilles_common_essentials.csv
-  # ctrl vs gene labels are from inst/extdata/pgPEN_annotations.txt
-
-  # This is the core of the code we'll need but we'll need to refactor
-  #d.annot <- d.annot %>%
-  #mutate(norm_ctrl_flag = case_when(
-  #  target_type == "gene_gene" ~ "double_targeting",
-  #  target_type == "gene_ctrl" & gene1_essential_flag == TRUE ~ "positive_control",
-  #  target_type == "ctrl_gene" & gene2_essential_flag == TRUE ~ "positive_control",
-  #  target_type == "gene_ctrl" & gene1_essential_flag != TRUE ~ "single_targeting",
-  #  target_type == "ctrl_gene" & gene2_essential_flag != TRUE ~ "single_targeting",
-  #  target_type == "ctrl_ctrl" ~ "negative_control")) %>%
-  #mutate(norm_ctrl_flag = factor(norm_ctrl_flag, levels = c("negative_control",
-  #                                                          "positive_control",
-  #                                                          "single_targeting",
-  #                                                          "double_targeting")))
-
-  gimap_dataset$annotation <- NULL #TODO: Final step is annotations that line up to the same order as the pg gene data should be stored here.
+  ############################ CONTROL GENE ANNOTATION #########################
+  # If control genes aren't provided then we get some from DepMap
+  if (!is.null(control_genes)) {
+    if (!file.exists(control_genes)) stop("The annotation_file specified cannot be found. Please double check the file path")
+    control_genes <- read_table(control_genes)[, 1]
+  } else {
+    # This file is from https://depmap.org/portal/download/all/ and from DepMap Public 19Q3 All Files
+    # Essential gene labeling is from inst/extdata/Achilles_common_essentials.csv
+    control_genes <- readr::read_tsv("https://figshare.com/ndownloader/files/40448429", show_col_types = FALSE)
+    control_genes <- control_genes %>%
+      tidyr::separate(col = Gene, into = c("gene_symbol", "entrez_id"), remove = FALSE, extra = "drop") %>%
+      dplyr::pull(gene_symbol)
+  }
+
+  ############################ Get TPM data ####################################
+  # This is not optional because its used to flag things
+  ## get TPM and CN information (w/ option for user to upload their own info)
```

Comment:
I don't see the options for users to upload their own info for TPM or CN annotation

Reply:
Happy to have a longer convo about this. Definitely open to making things more customizable later -- or now if its an easy add. 

Let me know if you have vision for what this could look like. Happy to explore that.

Reply:
> I appreciate this line of thinking youre going down. Definitely agree that having things parallel generally makes more sense for usage.
> 
> We could allow for this but at a certain point if _all_ the files are custom it might be something easier for user to just use a dplyr::join on their own. Functionalizing won't be very helpful if every file is custom.

Completely agree with this comment that if everything is custom, it might be easier for users to do the join themselves. 

I was marking these specific lines just to say I didn't see the option, so I would remove that part of the comment. 

I think the parallelization I would most want as a user is to save every annotation file in `extdata` if we're going to saves some of them